### PR TITLE
Refactor listen.py

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below
no changelog

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This should be a straight refactor, no changed logic. It moves a ton of business logic out of listen.py and into smaller files, which should be easier to navigate and reason about

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:23defb4-nikolaik   --name openhands-app-23defb4   docker.all-hands.dev/all-hands-ai/openhands:23defb4
```

### CODE REVIEW

File: openhands/server/middleware.py
Comment (amanape) [inline openhands/server/middleware.py:1]: Worth considering breaking it down further, e.g.,

```py
async def __call__(self, request: Request, call_next: Callable):
    if not self.should_attach(request):
        return await call_next(request)

    if not await self.verify_authentication(request):
        return JSONResponse(
            status_code=status.HTTP_401_UNAUTHORIZED,
            content={'error': 'Authentication failed'},
        )

    request.state.sid = self.extract_session_id(request)
    if not request.state.sid:
        return JSONResponse(
            status_code=status.HTTP_401_UNAUTHORIZED,
            content={'error': 'Invalid session ID'},
        )

    request.state.conversation = await self.attach_conversation(request.state.sid)
    if request.state.conversation is None:
        return JSONResponse(
            status_code=status.HTTP_404_NOT_FOUND,
            content={'error': 'Session not found'},
        )

    try:
        response = await call_next(request)
    finally:
        await self.detach_conversation(request.state.conversation)

    return response
```

File: openhands/server/routes/auth.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        'exp': int(time.time()) + 3600,  # 1 hour expiration
+    }
+    signed_token = sign_token(cookie_data, config.jwt_secret)
+
+    response = JSONResponse(
+        status_code=status.HTTP_200_OK, content={'message': 'User authenticated'}
+    )
+
+    # Set secure cookie with signed token
+    response.set_cookie(
```
Comment (amanape) [inline openhands/server/routes/auth.py:92]: Are we actually handling anything with this cookie in other requests?

File: openhands/server/routes/conversation.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+from openhands.runtime.base import Runtime
+
+app = APIRouter(prefix='/api')
+
+
+@app.get('/conversation')
+async def get_remote_runtime_config(request: Request):
+    """Retrieve the remote runtime configuration.
+
+    Currently, this is the runtime ID.
```
Comment (amanape) [inline openhands/server/routes/conversation.py:14]:
```suggestion
    """Retrieve the runtime configuration.

    Currently, this is the session ID and runtime ID (if available).
```

**Review by amanape — APPROVED** [review]
Beautiful!

File: openhands/server/middleware.py
Comment (rbren) [reply openhands/server/middleware.py:1]: hmm good idea--going to punt on this to minimize the diff though

**Pull request merged**